Prevented saving a file with an empty database;

cage2 opened the pull request:
Hi!

This patch try to address #104.

The general idea is (in function `annotate-dump-annotation-data`) to prevent the annotation file to be saved, or delete it, if the annotation database is empty (i.e. there are not buffers with annotated text when  `annotate-dump-annotation-data` is called).

Bye!
C.

Commit:
- prevented saving a file whith an empty database;

bastibe commented on annotate.el:
```diff
@@ -1565,19 +1572,32 @@ annotation."
+              (delete-file annotate-file t)
+            (error (message "error removing annotation database: %S"
+                            (error-message-string err))))
+        (annotate-dump-annotation-data data t)))))
```
Why the additional recursion here (and the new optional argument)? I'm curious: did you run into a case where you were able to write data to the file, but not delete it?

cage2 replied:
Hi!

The  recursive  call  is  made  when  the  user  totally  deleted  the
annotations in all  their files but asked not to  delete the database file,
in this case an empty (contains only  'nil') db is saved, this is the old
behaviour actually.

Or at least, this are the intentions of the code i wrote ;-) but seems
to work in my testing.

Bye!
C.

cage2 replied:
Hi!

The  recursive  call  is  made  when  the  user  totally  deleted  the
annotations in all  their files but asked not to  delete the database file,
in this case an empty (contains only  'nil') db is saved, this is the old
behaviour actually.

Or at least, this are the intentions of the code i wrote ;-) but seems
to work in my testing.

Bye!
C.

bastibe commented:
Looks reasonable.

cage2 commented:
Hi!

> Why the additional  recursion here (and the  new optional argument)?
> I'm curious: did  you run into a  case where you were  able to write
> data to the file, but not delete it?

The  recursive  call  is  made  when  the  user  totally  deleted  the
annotations in all  their files but asked not to  delete the database,
in this case an empty (contains only  'nil') is saved, this is the old
behaviour actually.

Or at least, this are the intentions of the code i wrote ;-) but seems
to work in my testing.

Bye!
C.

cage2 commented:
Hi!

> Looks reasonable.

Good! Honestly i did not like the idea to delete entirely the file but
the motivations  suggested in the report  was valid to me  so changing
the  managing of  empty database  was  the right  thing to  do, in  my
opinion; i am  happy to know that  you also think the  changes this PR
introduce are good! Now I would  wait some times (a week?) waiting for
feedback from  the user that  opened the issue report  before merging
this, do you think is it OK?

Bye!
C.

indigoviolet commented:
@cage2 Thank you for your work on this change. I pulled and tested it and it seems to work perfectly!

cage2 commented:
Hi!

> @cage2 Thank you for your work on this change. I pulled and tested it and it seems to work perfectly!

That's wonderful! I need to change the accessory files and version and i am going to merge this PR.

Bye and thanks for your help and suggestions!
C.

Commit:
- updated docs and version number.

cage2 commented:
Hi @bastibe !

I have updated version to 1.3.0 as there is a non trivial changes in the program behaivour here.
Also i have added just me to the NEWS file, but if you want to be added tour name too, you know i am totally OK with that!

Bye!
C.

bastibe commented:
Great work, as always!

cage2 commented:
Hi @bastibe !

Thank you!! :)

Merging in a sec! :)

Bye!
C.
